File: content-solutions/hash/magazine-letter.py
def solution(letters, message):
    have = {}
    for ch in letters:
        have[ch] = have.get(ch, 0) + 1
    for ch in message:
        if have.get(ch, 0) == 0:
            return False
        have[ch] -= 1
    return True

File: content-solutions/hash/magazine-letter.java
import java.util.*;

class Solution {
    public boolean solution(String letters, String message) {
        Map<Character, Integer> have = new HashMap<>();
        for (char ch : letters.toCharArray()) have.put(ch, have.getOrDefault(ch, 0) + 1);
        for (char ch : message.toCharArray()) {
            if (have.getOrDefault(ch, 0) == 0) return false;
            have.put(ch, have.get(ch) - 1);
        }
        return true;
    }
}
